chain/network/src/peer_manager/tests/nonce.rs

```diff
@@ -75,3 +78,57 @@ async fn test_nonces() {
         }
     }
 }
+
+async fn wait_for_edge(actor_handler: &mut ActorHandler) -> Edge {
+    actor_handler
+        .events
+        .recv_until(|ev| match ev {
+            Event::PeerManager(peer_manager_actor::Event::EdgesVerified(ev)) => Some(ev[0].clone()),
+            _ => None,
+        })
+        .await
+}
+
+#[tokio::test]
+/// Create 2 peer managers, that connect to each other.
+/// Verify that the will refresh their nonce after some time.
+async fn test_nonce_refresh() {
+    init_test_logger();
+    let mut rng = make_rng(921853255);
+    let rng = &mut rng;
+    let mut clock = time::FakeClock::new(*EDGE_MIN_TIMESTAMP_NONCE + time::Duration::days(2));
```

Comment:
you might want to call clock.set_autoadvance(Duration::ZERO) if you want to strictly control the time, because as of today the clock is autoadvancing at each call to clock.now/now_utc().

Reply:
do we have such option in our fake clock? as I didnt' see it.

Reply:
yeah, sorry, it was not merged yet at the time of writing this comment, and I already removed it in favor of proper sleep() support.